### PR TITLE
Fix null pointer in tracing on javascript

### PR DESCRIPTION
**Pull request by Jasper-M**
As seen here: https://discord.com/channels/632277896739946517/632278585700384799/1062698867172311090

### CODE REVIEW

**Comment by armanbilge** [comment]
Thanks! Instead of using `Option` here, what would you think about just doing a `null` check? It's not nice code, but it avoids the allocation and will be better for performance.

File: core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala [hunk 1]
```diff
@@ -104,11 +104,12 @@ private[tracing] abstract class TracingPlatform { self: Tracing.type =>
   private[tracing] def applyStackTraceFilter(
       callSiteClassName: String,
       callSiteMethodName: String,
-      callSiteFileName: Option[String]): Boolean = {
+      callSiteFileName: String): Boolean = {
 
     // anonymous lambdas can only be distinguished by Scala source-location, if available
     def isInternalScalaFile =
-      callSiteFileName.exists(fileName => !fileName.endsWith(".js") && isInternalFile(fileName))
+      callSiteFileName != null && !callSiteFileName.endsWith(".js") && isInternalFile(
```
Comment (armanbilge) [inline core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala]:
```suggestion
      callSiteFileName ne null && !callSiteFileName.endsWith(".js") && isInternalFile(
```

File: core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala [hunk 2]
```diff
@@ -107,8 +107,9 @@ private[tracing] abstract class TracingPlatform { self: Tracing.type =>
       callSiteFileName: String): Boolean = {
 
     // anonymous lambdas can only be distinguished by Scala source-location, if available
-    def isInternalScalaFile = !callSiteFileName.endsWith(".js") &&
-      isInternalFile(callSiteFileName)
+    def isInternalScalaFile =
+      callSiteFileName ne null && !callSiteFileName.endsWith(".js") && isInternalFile(
```
Comment (armanbilge) [inline core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala]:
```suggestion
      (callSiteFileName ne null) && !callSiteFileName.endsWith(".js") && isInternalFile(
```

File: core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala (hunk 1 above)
Comment (Jasper-M) [reply core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala]: The compiler emits the same bytecode for both 😛 I hope Scala.js does the same.

Comment (armanbilge) [reply core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala]: > I hope Scala.js does the same.

Seems not:

```
  $dp_equals__O__Z($m_Ltest$package$().whatever__O(), null);
  ($m_Ltest$package$().whatever__O() !== null)
```

Comment (armanbilge) [reply core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala]: Neither does Scala JVM.

```
this.whatever().equals(null);
this.whatever() != null;
```

> The compiler emits the same bytecode for both

What is this based on? 🤔

Comment (Jasper-M) [reply core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala]: I tested in the REPL
```scala
scala> :paste -raw
// Entering paste mode (ctrl-D to finish)

class Foo{ def foo(a: String) = a != null; def bar(a: String) = a ne null }

// Exiting paste mode, now interpreting.

scala> :javap -cp Foo
Compiled from "<pastie>"
public class Foo {
  public boolean foo(java.lang.String);
    Code:
       0: aload_1
       1: ifnull        8
       4: iconst_1
       5: goto          9
       8: iconst_0
       9: ireturn

  public boolean bar(java.lang.String);
    Code:
       0: aload_1
       1: ifnull        8
       4: iconst_1
       5: goto          9
       8: iconst_0
       9: ireturn
```

Comment (armanbilge) [reply core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala]: sorry, you are completely right!! I think I was doing something dumb

Comment (Jasper-M) [reply core/js/src/main/scala/cats/effect/tracing/TracingPlatform.scala]: Doesn't matter 😛 your change makes sense either way. I wasn't completely sure myself.

**Pull request merged**